APE26

jeffjennings opened the pull request.

astrofrog committed:
Use wcsenq to check for changes in wcsprm to determine whether wcsset needs to be called

astrofrog committed:
Added changelog entry

neutrinoceros committed:
API: make delay_doc_updates context manager a noop under Python's optimized mode

jeffjennings commented on astropy/coordinates/abstract.py:
```diff
@@ -0,0 +1,1550 @@
+            "Mapping for frame-specific component names",
+        )
+
+        # TODO APE: keep?
```
@adrn @mhvk @nstarman - do we need this?

nstarman replied:
I would say not in `AbstractCoordinate`, but yes in `SkyCoord`.

jeffjennings commented on astropy/coordinates/abstract.py:
```diff
@@ -0,0 +1,1550 @@
+        self._representation = self._infer_representation(
+            representation_type, differential_type
+        )
+        # TODO APE: can we simplify this?
```
@adrn @mhvk @nstarman - should we refactor/simplify?

jeffjennings commented on astropy/coordinates/abstract.py:
```diff
@@ -0,0 +1,1550 @@
+            # likely a SkyCoord object.
+            new_frame = new_frame._sky_coord_frame
+
+        # TODO APE: revisit and potentially refactor
```
@adrn @mhvk @nstarman - should we refactor this?

jeffjennings commented on astropy/coordinates/abstract.py:
```diff
@@ -0,0 +1,1550 @@
+        data_repr_spl[-1] = "\n".join((first, *middle, last.removesuffix(">")))
+        return "\n".join(data_repr_spl)
+
+    # TODO APE: revisit and potentially refactor
```
@adrn @mhvk @nstarman - should we refactor this?

neutrinoceros committed:
TST: narrow a warning catch in `coordinates`' test suite (fix `PT030`)

astrofrog committed:
Added ``preserve_units`` to WCS class to allow units to be preserved as far
as the user is concerned rather than have them be converted to SI (for example
converting nm to m, GHz to Hz, and arcsec to deg).

astrofrog committed:
Fix missing Python <-> C conversion

astrofrog committed:
Fix function name

astrofrog committed:
Fix setting cdelt and crval

astrofrog committed:
Don't allow cunit to be changed in WCS when using preserve_units=True after set() has been called

astrofrog committed:
Fix getting/setting CD matrix, and disable setting individual items

astrofrog committed:
Add test of to_header

astrofrog committed:
Added WCSAxes figure test and fix default format_unit for RA if world unit is not degrees

astrofrog committed:
Fix case where set() is called explicitly

astrofrog committed:
Test more cases of programmatic use

astrofrog committed:
Added more tests

astrofrog committed:
Added figure hash

astrofrog committed:
Skip codespell for test file

eerovaher committed:
Simplify `utils.metadata.merge()`

astrofrog committed:
Fix str, repr, and print_contents

astrofrog committed:
Fix codestyle

astrofrog committed:
Added regression test

astrofrog committed:
Fix tolerance in test

astrofrog committed:
Document test and fix code style

astrofrog committed:
Added news fragment

astrofrog committed:
Fixed bugs with WCS slicing, specifically that slicing WCS objects with more than 2 dimensions but with _naxis not set would fail, and that slicing a WCS with an ellipsis would not return a WCS object but a SlicedFITSWCS instead

astrofrog committed:
Fix codestyle

mcara committed:
Fix _naxis dimensionality

mcara committed:
Remove @astrofrog code

mcara committed:
Add change log entry

mcara committed:
fix eol

Nanoemc committed:
Fixed problems with unsupported numpy keys for certain numpy functions

Nanoemc committed:
Adding changelog docs

Nanoemc committed:
Added tests for ValueError exceptions with keys

mwcraig committed:
Merge pull request #18424 from Nanoemc/issue_15397

FIX: Issue 15397

astrofrog committed:
Merge pull request #18419 from mcara/fix_naxis

Fix _naxis dimensionality

astrofrog committed:
Remove fix which is no longer needed

astrofrog committed:
Updated changelog entry

eerovaher committed:
Deprecate unused `matrix_utilities.is_rotation()`

The function is not documented, but it is nonetheless mentioned in the
change log, so it is deprecated as a courtesy instead of being removed
immediately.

eerovaher committed:
Deprecate unused `matrix_utilities.angle_axis()`

The function is not documented, but it is nonetheless mentioned in the
change log, so it is deprecated as a courtesy instead of being removed
immediately.

mcara committed:
Merge pull request #18417 from astrofrog/fix-wcs-slicing

Fixed bugs with WCS slicing,

ayshih committed:
Add a test for the shape of a 1D WCS

pllim committed:
Merge pull request #18405 from ayshih/wcs_1d

Add a test for the reporting of the shape of a 1D WCS

eerovaher committed:
Make .py files without shebang not executable

Ruff rule EXE002 (shebang-missing-executable-file) indentifies
executable Python files that nonetheless have no shebang. There is no
reason why text files without a shebang should be executable.

astrofrog committed:
Improvements to C code, and expand tests to use a wider range of input values

astrofrog committed:
Fix codestyle and test

astrofrog committed:
Add missing return

astrofrog committed:
Fix C warnings

astrofrog committed:
Output CUNIT in WCS __repr__, and add documentation and what's new entry

astrofrog committed:
Fix rst syntax

astrofrog committed:
Fix doctests and remove trailing spaces in WCS.__repr__

astrofrog committed:
Update test

astrofrog committed:
Fixed WCS.__repr__

pllim committed:
Merge pull request #18434 from eerovaher/ruff-EXE002

Ensure Python files without a shebang are not executable

astrofrog committed:
Avoid modifying input world coordinates in-place

pllim committed:
Merge pull request #18301 from neutrinoceros/table/tst/refine-too-broad-pytest.warns

TST: narrow a warning catch in `table`'s test suite (fix `PT030`)

astrofrog committed:
Added regression test for in-place modifications to world array, and started adding -TAB test

astrofrog committed:
Fixed code style

astrofrog committed:
Merge pull request #18324 from astrofrog/warnings-invalid-axes

Emit warning if axes given to set_*_position are invalid

taldcroft committed:
Check ruff B007 in time and fix issues

taldcroft committed:
Merge pull request #18295 from neutrinoceros/table/rfc/ruf058

RFC: auto-fix unique violation of lint rule `RUF058` in `table`

taldcroft committed:
Merge pull request #18298 from neutrinoceros/io.ascii/tst/refine-too-broad-pytest.warns

TST: narrow a warning catch in `io.ascii`'s test suite (fix `PT030`)

pllim committed:
Merge pull request #18438 from taldcroft/ruff-time-b007

Enable ruff B007 (unused control loop variable) in time

astrofrog committed:
Fix failing test

pllim committed:
Merge pull request #18441 from astrofrog/fix-wcsaxes-ci

Fix failing test in astropy.visualization.wcsaxes

astrofrog committed:
Fix test_tab

astrofrog committed:
Rename test

astrofrog committed:
Add a private property to check unit scaling, and only do scaling if any of the scales are not 1. Also explicitly check the units of the TAB WCS

astrofrog committed:
Fixed tests

tomdonaldson committed:
Merge pull request #18366 from timj/u/timj/votable-config

Try to improve version checks in VOTable

astrofrog committed:
Fix error handling for setting cunit

eerovaher committed:
Introduce `coordinates.is_rotation_or_reflection()`

The new function is essentially `matrix_utilities.is_O3()` renamed. The
old name was obscure to anyone not familiar enough with group theory to
recognize the O(3) group. The function is not documented under its old
name, but it is nonetheless mentioned in the change log, so the old name
is deprecated as a courtesy instead of being removed immediately.

eerovaher committed:
Make matrix utilities available from `coordinates`

`is_rotation_or_reflection()` and `rotation_matrix()` are now available
from the top-level `coordinates` namespace.

`matrix_transpose()` is not made available from `coordinates` because
users more concerned about readability can use `np.matrix_transpose()`
(starting from `numpy` 2.0) and users more concerned about performance
should use the `swapaxes()` method directly.

The other matrix utilities are deprecated.

pllim committed:
DOC: Add known issue for np.prod on Quantity

mhvk committed:
Merge pull request #18449 from pllim/np-prod-iykyk

DOC: Add known issue for np.prod on Quantity

preshanth committed:
Dark Energy Component Trait Class (#18447)

* Initial commit of darkenergycomponent trait class
* Updating docs and changelog files
* Adding dark energy to public api

Co-authored-by: Nathaniel Starkman <nstarman@users.noreply.github.com>

astrofrog committed:
Started adding support for changing cunit

astrofrog committed:
Make it so that units can be changed on a WCS even if using preserve_units=True

astrofrog committed:
Raise error if trying to modify  units in-place after set() has been called when using preserve_units=True, and also make sure we correctly raise errors related to incompatible units

astrofrog committed:
Simplify implementation of set_cunit

eerovaher committed:
Speed up `utils.diff.diff_values()` (#18446)

* Add a new test case to `diff_values()` tests

* Speed up `astropy.utils.diff.diff_values()`

The function can be sped up for scalar `float` inputs by calling
standard library `math` functions instead of `numpy`.

E-W-Jones committed:
Fix typo in FLRW Ode method docstring (#18314)

* Fix typo in DarkEnergyComponent.Ode method. There was a typo in the FLRW Ode method, saying it returned non-relativistic matter density instead of dark energy.

eerovaher committed:
Fix a pickling test with spurious xfails

neutrinoceros committed:
WHL: test wheels against CPython 3.14 (GIL-enabled)

neutrinoceros committed:
TST: remove macOS + x86_64 combo from weekly cron

astrofrog committed:
Merge pull request #18338 from astrofrog/wcs-units-c

Implement option in WCS class to preserve the original units

pllim committed:
Merge pull request #18460 from neutrinoceros/tst/drop-native-mac-x86_64

TST: remove macOS + x86_64 combo from weekly cron

eerovaher committed:
Remove unused `ErfaAstromInterpolator._get_c2i()`

The function has served no purpose since
adcf09b13984d177e949a372c83d27f5b1d242a1 (remove apci)

astrofrog committed:
Explicitly list all imported objects from the astropy.wcs._wcs C extension, and remove fallbacks if extension can't be imported

astrofrog committed:
Rename _wcs.__version__ to _wcs.WCSLIB_VERSION

astrofrog committed:
Fix codestyle

mhvk committed:
Merge pull request #18464 from eerovaher/rm_get_c2i

Remove unused `ErfaAstromInterpolator._get_c2i()`

mhvk committed:
Merge pull request #18291 from eerovaher/prevent-PLR1733

Simplify `utils.metadata.merge()`

astrofrog committed:
Merge pull request #18457 from neutrinoceros/whl/test-wheels-on-cp314

WHL: test wheels against CPython 3.14 (GIL-enabled)

mcara committed:
Merge pull request #18466 from astrofrog/astropy-wcs-import

Clean up setup of astropy.wcs module

astrofrog committed:
C formatting

Co-authored-by: Mihai Cara <mcara@users.noreply.github.com>

astrofrog committed:
Merge pull request #16411 from astrofrog/avoid-unecessary-wcsset

Avoid calling wcsset if the wcsprm is unchanged

saimn committed:
allow setting EXTNAME when writing a FITS Table

eerovaher committed:
Return NamedTuples from coordinate match functions

The coordinate matching and searching functions and the corresponding
`SkyCoord` methods have so far returned tuples. Returning instances of
new `NamedTuple` subclasses allows accessing values in the returned
tuples via named attributes without breaking backwards compatibility.

pllim committed:
Merge pull request #18470 from saimn/table-extname

Allow setting EXTNAME when writing a FITS Table

dependabot[bot] committed:
Bump the actions group in /.github/workflows with 2 updates

Bumps the actions group in /.github/workflows with 2 updates: [OpenAstronomy/github-actions-workflows](https://github.com/openastronomy/github-actions-workflows) and [actions/first-interaction](https://github.com/actions/first-interaction).


Updates `OpenAstronomy/github-actions-workflows` from 2.1.0 to 2.2.1
- [Release notes](https://github.com/openastronomy/github-actions-workflows/releases)
- [Commits](https://github.com/openastronomy/github-actions-workflows/compare/86823142467dd2afcd1bbac57d3a664cf468eb3b...28e947497bed4d6ec3fa1d66d198e95a1d17bc63)

Updates `actions/first-interaction` from 1.3.0 to 2.0.0
- [Release notes](https://github.com/actions/first-interaction/releases)
- [Commits](https://github.com/actions/first-interaction/compare/34f15e814fe48ac9312ccf29db4e74fa767cbab7...2d4393e6bc0e2efb2e48fba7e06819c3bf61ffc9)

---
updated-dependencies:
- dependency-name: OpenAstronomy/github-actions-workflows
  dependency-version: 2.2.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: actions
- dependency-name: actions/first-interaction
  dependency-version: 2.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: actions
...

Signed-off-by: dependabot[bot] <support@github.com>

taldcroft committed:
Update the RDB format description

larrybradley committed:
Merge pull request #18453 from eerovaher/fix-pickling-test

Fix pickling tests with spurious xfails in `convolution`

taldcroft committed:
Provide better visibility for Learn Astropy and Coordinated/Affiliated package in core docs (#18409)

* WIP to update index page to highlight learn and coordinated

* Tighten up cards and make this ready for a PR

* Add learn and packages icons

* Update docs/index.rst

Co-authored-by: P. L. Lim <2090236+pllim@users.noreply.github.com>

---------

Co-authored-by: P. L. Lim <2090236+pllim@users.noreply.github.com>

pllim committed:
Merge pull request #18485 from taldcroft/docs-rdb-format-description

Update the RDB format description

pllim committed:
Merge pull request #18484 from astropy/dependabot/github_actions/dot-github/workflows/actions-9be1d38f42

Bump the actions group in /.github/workflows with 2 updates

pllim committed:
Merge pull request #18418 from eerovaher/rm-is_rotation

Sort out public and private matrix utilities

eerovaher committed:
Remove support for `h5py` older than 3.9

Older versions of `h5py` used `numpy.product()`, which was deprecated in
`numpy` 1.25. Not supporting old `h5py` prevents deprecation warnings
with most supported `numpy` versions.

invalid-email-address committed:
Update minimum required version of astropy-iers-data

pllim committed:
Merge pull request #18493 from astropy/update-astropy-iers-data-pin-1754093735

Update minimum required version of astropy-iers-data

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.12.2 → v0.12.7](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.2...v0.12.7)

eerovaher committed:
Merge pull request #18496 from astropy/pre-commit-ci-update-config

[pre-commit.ci] pre-commit autoupdate

bsipocz committed:
MAINT: minor cleanup of duplicate check

pllim committed:
Merge pull request #18497 from bsipocz/MAINT_minor_cleanup

MAINT: minor cleanup of duplicate check

larrybradley committed:
Future proof skipping building free-threaded wheels

pllim committed:
Merge pull request #18499 from larrybradley/freethreaded-wheels

Future proof skip for building free-threaded wheels

hamogu committed:
Explain why HTML readers might not be good choice for journal tables

In #14346 it was noted that the HTML table reader does not easily return a table from an MNRAS article. Deeper study shows that there are problems with any major journal, due to inconsistent mark-ups, captchas, changing formats, embedded JS, or that tables are only included as jpegs in some cases. Thus, it seems impossible to make a working table parser for even the top few journals (see details of discussion in  #14346).
This PR tries ot summarize the findings in the API and narrative docs in the locations that I hope users are most likely to see it. The discussion of the original issue suggested more extreme measures (such as rasing a warning every time the HTML reader is used or deprecating it altogether) but I deem that too invasive, since the reader itself works and is useful for a number of legitimate cases, including the case where a user manually downloads the HTML for an article and reads it after verifying that the table in question is actually contained in the file.

closes #14346

[skip CI]

eerovaher committed:
Add a test for formatting the liter

The new test demonstrates that "ℓ" is being used as the symbol for the
liter with some unit formatters, but according to the SI brochure
(https://www.bipm.org/documents/20126/41483022/SI-Brochure-9-EN.pdf) the
symbols for the liter are "l" or "L".

eerovaher committed:
Ensure the liter is always represented as `"l"`

Some unit formatters represented the liter as `"ℓ"` and others as `"l"`.
According to the SI brochure
(https://www.bipm.org/documents/20126/41483022/SI-Brochure-9-EN.pdf) the
symbol should be `"l"` or `"L"`, so the formatters that have so far used
`"ℓ"` now use `"l"` to be consistent with the others.

pllim committed:
Merge pull request #18501 from hamogu/docs_html_reader

Explain why HTML readers might not be good choice for journal tables

pllim committed:
Merge pull request #18492 from eerovaher/h5py-3.9

Remove support for `h5py` older than 3.9

eerovaher committed:
Add a test to reveal a bug in `rotation_matrix()`

According to the documentation the `angle` argument of
`coordinates.rotation_matrix()` can be an angle-like, but the new test
reveals that angle-like strings are nonetheless rejected.

eerovaher committed:
Fix a bug about `rotation_matrix()` `angle` types

According to the docstring of `rotation_matrix()` its `angle` argument
can be any angle-like value, but since
63369050f360419214548f3f5ab4f48510af11b2 (Improve performance of
rotation_matrix for common angle units) and
a4dc7bc701b11a38e556b35736c3998cb0aaa83c (Bug fix and suggested
improvements) it has rejected some angle-like types, e.g. angle-like
strings. The new code fixes the bug and retains a shortcut for
`Quantity` (and `Angle`) instances.

mhvk committed:
Merge pull request #18504 from eerovaher/rotation_matrix-input-types

Allow all angle-like `angle` arguments in `rotation_matrix()`

hamogu committed:
Improve error message when column formatting fails (#17631)

* Improve error message when column formatting fails

Add additional details to the error message when setting the format of a table column fails.
The message is not complete in the sense that it does not point the user to all possible settings (e.g. old-style formatting is also allowed), but it should set expectations right and make clear that the `.format` is just for printing and will not change the dtype of a column (see #8071).

* Add changelog

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Break line and change URL as suggest in code review

* Update astropy/table/pprint.py

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

hamogu committed:
Find metadata in CDS readme even for gzipped files

CDS/Vizier serves large datafiles in gzipped for by default, but the ".gz" ending is not included in the filename in the ReadMe file and thus the reader will not find the metadata and fail.
This PR simply strips off any ".gz" ending before searching the ReadMe.

closes #6549

taldcroft committed:
Add new ECSV reader allowing different engines for CSV reader (#18267)

* Initial implementation of pyarrow engine for ECSV

* Working for simple test cases

* Passing minimally mod'd io.ascii ECSV tests

* Allow for different null values and compressed files

* Change ascii.read to Table.read where possible

* Mostly working refactor of type handling

* Refactor passing tests

* Baseline update io.ascii tests for both pyarrow and io.ascii engines (not passing)

* All tests except float16 passing

* Mostly complete refactor

* Still more refactoring

* Remove unnecessary default setting for dtype

* Work around PyArrow not supporting float16

* Introduce engine arg

* Use `engine` with new "ecsv" format

* Improve fill_values handling to pass tests

* Make SerializedColumn repr more helpful for debugging

* Add support for Pandas nullable boolean

* Add pandas.csv engine for ECSV reader

Passing all tests except one.

* Start refactoring into ECSVEngine classes

* More refactoring

* Fix issue with float nan/NaN with pandas engine

* Docs, name changes

* Tiny docstring mistake

* More name changes

* More renaming

* Separate engine name and format.

Plus more docs improvements.

* Typing and doc updates

* Move new ecsv with engines out of pyarrow

* Remove orphaned test file

* Fix ruff

* Add DerivedColumnProperties NamedTuple subclass

* Use slots=True, frozen=True for ECSVHeader dataclass

* Docstring update

* WIP using instances for ECSV engine not classes

* Instance based passing tests including new one

* Adjust tests based on optional dependencies

* Raise TypeError for float128 on pyarrow

* Expose new ecsv module documentation and address some review comments

* Address more review comments

* Fix circular dependencies for `import astropy.io.misc.ecsv`

* Changes to allow frozen=True for ColumnECSV dataclass

* Change ColumnECSV.dtype property to be numpy.dtype

* Fix typing and documentation to resolve confusion about str_vals (always 1-d)

* Address further review comments

* Add changelog

* Documentation updates

eerovaher committed:
Prefer module level imports in `coordinates`

mhvk committed:
Merge pull request #18507 from eerovaher/coordinates-module-level-imports

Prefer module level imports in `coordinates`

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mhvk committed:
Merge pull request #18500 from eerovaher/liter-str

Make the string representation of the liter consistent between different unit formatters

mohsinm-dev committed:
Update IVOA UCD1+ controlled vocabulary from version 1.5 to 1.6 (#18483)

* Update IVOA UCD1+ controlled vocabulary from version 1.5 to 1.6

This update adds support for new atmospheric observation terms including
obs.atmos.wind, obs.atmos.humidity, obs.atmos.rain, obs.atmos.turbulence,
obs.atmos.turbulence.isoplanatic, and obs.atmos.water which are now
recognized when parsing UCDs with check_controlled_vocabulary=True.

Fixes #18452

* Fix formatting issues in UCD test

- Remove trailing whitespace and blank line whitespace
- Fix PLR0133 by using variable instead of inline string comparison
- Ensure proper line formatting per ruff standards

* Fix formatting and line ending issues

- Apply ruff formatting to test file
- Add missing newline to changelog file
- Ensure all pre-commit checks will pass

* Fix test_atmospheric_wind_ucd to properly handle UCD word types

Secondary words (obs.atmos.wind, obs.atmos.turbulence) can only be used
after primary words, while quantified/primary words can be used standalone.
The test now correctly separates these cases and tests secondary words
only in valid secondary positions.

* Apply ruff formatting to test_ucd.py

* Address PR feedback: remove has_colon parameter and add phys.temperature.dew

* Fix trailing whitespace in changelog

eerovaher committed:
Remove return statements from __init_subclass__()

The only value a `__init_subclass__()` method should return is `None`
and that can be done implicitly.

pllim committed:
DOC: quantity_support after import
in narrarive doc to avoid user confused by ruff complain when copy-paste

larrybradley committed:
Merge pull request #18513 from pllim/reorder-mpl-import

DOC: quantity_support after import in narrarive doc

mhvk committed:
Merge pull request #18459 from eerovaher/coordinate-matching-return-types

Return NamedTuples from coordinate match functions

pllim committed:
Merge pull request #18512 from eerovaher/__init_subclass__-return

Remove needless `return` statements from `__init_subclass__()`

eerovaher committed:
Remove two `ErfaAstromInterpolator` static methods

The removed static methods were only called by the `apco()` method and
their functionality is simpler to achieve using a local function defined
with `functools.partial()`.

mhvk committed:
Merge pull request #18469 from eerovaher/ErfaAstromInterpolator-staticmethods

Eliminate two private `ErfaAstromInterpolator` static methods

neutrinoceros committed:
CLN: simplify argument parser settings

eerovaher committed:
Move duplicated code to a new function

hamogu committed:
Activate as many doctests and feasible in narrative unified I/O interface

doctests with dependencies (hdf5 and parquet) are set up, but still skipped due to
https://github.com/scientific-python/pytest-doctestplus/issues/291

hamogu committed:
Activate doctests for hdf5 and parquet with individual `doctest-requires` guards

closes #13265

hamogu committed:
Use more files included in astropy for examples instead of secretly generated files

pllim committed:
Merge pull request #18520 from neutrinoceros/cln/simplify-argparse-settings

CLN: simplify argument parser settings

hamogu committed:
Add a new Time ``GalexSec`` for the GALEX satelite (#18330)

* Add a new Time ``GalexSec`` for the GALEX satelite

We've had Chandta in astropy for a long time, but I think it's time to add more observatories that use similar schemes - but with different scales and zeropoints.

* Address code review

- Update code comment about time accuracy
- Improvements for formatting and tests

Co-authored-by: Tom Aldcroft <taldcroft@gmail.com>

* Add additional cautious remarks on timing accuracy in multiple places

---------

Co-authored-by: Tom Aldcroft <taldcroft@gmail.com>

hamogu committed:
move doctest requirements to pytest.toml

eerovaher committed:
Add initial type annotations to coordinates/sites.py

mhvk committed:
Merge pull request #18529 from eerovaher/coordinates-sites-annotations

Add initial type annotations to `coordinates/sites.py`

taldcroft committed:
Fix bug removing rows on an indexed and sliced table (#18511)

* Fix bug removing rows on an indexed and sliced table

* Add another test case

* Add docs: AI-generated then updated by hand.

* Remove unnecessary hasattr check for col.info.indices

* Fix docstring in index.py

mhvk committed:
Merge pull request #18524 from hamogu/fix_13265

Activate doctests as far as feasible for narrative docs of unified I/O interface

mhvk committed:
Merge pull request #18522 from eerovaher/refco-helper

Simplify handling refraction coefficients

neutrinoceros committed:
DOC: add a badge with currently supported Python versions (as of the latest stable release)

neutrinoceros committed:
TST: avoid a deprecation warning from numpy 2.4 (`np.testing.suppress_warnings` is deprecated)

neutrinoceros committed:
Update README.rst

Co-authored-by: P. L. Lim <2090236+pllim@users.noreply.github.com>

pllim committed:
Merge pull request #18538 from neutrinoceros/timeseries/tst/avoid-numpy2.4-deprecation-warning

TST: avoid a deprecation warning from numpy 2.4 (`np.testing.suppress_warnings` is deprecated)

eerovaher committed:
Change the error for invalid site name types

The site registry (used e.g. in `EarthLocation.of_site()`) expects site
names to be strings. Previously if an incorrect type was passed in an
`AttributeError` was raised, but now a less surprising `TypeError` is
produced.

bhpcv252 committed:
document ability to explicitly set axis units

mhvk committed:
Merge pull request #18541 from eerovaher/site_name-type-error

Change the error for invalid site name types

taldcroft committed:
Fix bug where coordinate frame fails serializing to YAML (#18526)

* Fix bug where coord frame fails serializing to YAML

* Only support YAML for built-in frame classes

In particular exclude dynamically-created like SkyOffset ones like `abc.SkyOffset*`.

* Implement review comment and fix test for parallel

eerovaher committed:
Reveal a bad `UnitBase.to_string()` error message

A test was modified to reveal that if `UnitBase.to_string()` raises an
error then the message can contain irrelevant and misleading information
about available unit formatters.

eerovaher committed:
Clean up `UnitBase.to_string()` error messages

Error messages raised by `UnitBase.to_string()` no longer complain about
unit formatter unless an invalid formatter was the reason for the error.

mhvk committed:
Merge pull request #18545 from eerovaher/unit-to_string-error-msg

Remove clutter from `UnitBase.to_string()` error messages

mikez committed:
Add script to make type stubs for units (#18547)

eerovaher committed:
Avoid stringified type annotations in `units`

Sphinx is known to have problems resolving stringified annotations, so
it's best to avoid them whenever possible. In `units` it is not possible
to avoid stringified annotations entirely because of circular imports,
but it was nonetheless possible to remove `if TYPE_CHECKING:` blocks
from all but two modules. Moving the definitions of `UnitBase` and
`PhysicalType` was necessary for minimizing the number of forward
references.

mhvk committed:
Merge pull request #18329 from eerovaher/runtime-type-imports-units

Avoid stringified type annotations in `units`

mhvk committed:
Exclude structured_units.rst for erfa-dev (#18551)

* Exclude structured_units.rst for erfa-dev

* Update pyproject.toml [skip ci]

Co-authored-by: P. L. Lim <2090236+pllim@users.noreply.github.com>

---------

Co-authored-by: P. L. Lim <2090236+pllim@users.noreply.github.com>

pllim committed:
DOC: Fix wrong link in build from source
instructions by showing different text when not dev

pllim committed:
Merge pull request #18556 from pllim/fix-build-source-stable

DOC: Fix wrong link in build from source

larrybradley committed:
Merge pull request #18542 from bhpcv252/fix-18540

Document ability to explicitly set axis units with Matplotlib

eerovaher committed:
Improve duplicate unit definition error message

The machinery that injects units into a namespace checks for name
collisions. Because Python applies NFKC normalization to identifiers
then a name collision can occur even if before normalization the names
are different. The new error message is less cluttered if normalization
is not required for collision and more informative if it is.

mhvk committed:
Merge pull request #18559 from eerovaher/duplicate-unit-error-msg

Improve the error message in case of unit name collisions

eerovaher committed:
Show that unit modules insert bad names to __all__

A new test reveals that unit definition modules can insert names to
`__all__` that are not NFKC normalized. Such names are invalid because
Python automatically applies NFKC normalization to all identifiers.

eerovaher committed:
Avoid inserting non-NFKC unit names to __all__

The unit definition modules no longer insert names that are not NFKC
normalized into their `__all__` lists. Such names are invalid because
Python automatically applies NFKC normalization to all identifiers. This
also means that e.g. `from astropy import units as u; u.ℓ` still works
because Python converts the "ℓ" to "l", so this patch is fully
backwards-compatible.

mhvk committed:
Merge pull request #18563 from eerovaher/unit-module-dunder-all-nfkc

Stop inserting non-NFKC names to unit definition module `__all__` lists

casparvl committed:
Fixes test_fitting_with_initial_values[DogBoxLSQFitter] convergence failure on arm64 (#17958)

* Fixes  https://github.com/astropy/astropy/issues/15650 :test_fitting_with_initial_values[DogBoxLSQFitter] failure on arm64 when using an OpenBLAS optimized for ARM64 using -march=native

---------

Co-authored-by: Caspar van Leeuwen <casparl@tcn1.local.snellius.surf.nl>

mhvk committed:
Add changelog and whats-new entries

nstarman committed:
Merge pull request #18547 from mikez/main

Add script to make type stub for units.

mhvk committed:
Ensure no global changes to the warnings filter.

pllim committed:
Merge pull request #18568 from mhvk/units-stubs-warning-filter

Ensure no global changes to the warnings filter.

eendebakpt committed:
Replace deprecated setting of shape and dtype on numpy arrays (#18562)

* Use reshape instead of setting shape directly on numpy arrays

* Replace setting dtype with view()

mhvk committed:
ENH: Allow setup.py to create unit stubs on installation

This moves the previous script to a new file in typing_utils
and adjusts it so that it can be called from setup.py on
installation (in regular or editable mode). Test are adjusted
accordingly.

Since typing is at present not directly useful, and IDEs will
find these stubs automatically, no announcement of this is
made in the changelog or documentation.

mhvk committed:
MAINT: Ensure pyerfa is not needed for stubs building

Cadair committed:
Add a regression test for astropy/astropy#18576

Cadair committed:
Add Fix

dependabot[bot] committed:
Bump the actions group in /.github/workflows with 3 updates

Bumps the actions group in /.github/workflows with 3 updates: [actions/checkout](https://github.com/actions/checkout), [actions/cache](https://github.com/actions/cache) and [actions/first-interaction](https://github.com/actions/first-interaction).


Updates `actions/checkout` from 4.2.2 to 5.0.0
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/11bd71901bbe5b1630ceea73d27597364c9af683...08c6903cd8c0fde910a37f88322edcfb5dd907a8)

Updates `actions/cache` from 4.2.3 to 4.2.4
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/5a3ec84eff668545956fd18022155c47e93e2684...0400d5f644dc74513175e3cd8d07132dd4860809)

Updates `actions/first-interaction` from 2.0.0 to 3.0.0
- [Release notes](https://github.com/actions/first-interaction/releases)
- [Commits](https://github.com/actions/first-interaction/compare/2d4393e6bc0e2efb2e48fba7e06819c3bf61ffc9...753c925c8d1ac6fede23781875376600628d9b5d)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 5.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: actions
- dependency-name: actions/cache
  dependency-version: 4.2.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: actions
- dependency-name: actions/first-interaction
  dependency-version: 3.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: actions
...

Signed-off-by: dependabot[bot] <support@github.com>

Cadair committed:
Add changelog

neutrinoceros committed:
Merge pull request #18578 from astropy/dependabot/github_actions/dot-github/workflows/actions-9dff5394ab

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pre-commit/pre-commit-hooks: v5.0.0 → v6.0.0](https://github.com/pre-commit/pre-commit-hooks/compare/v5.0.0...v6.0.0)
- [github.com/woodruffw/zizmor-pre-commit: v1.11.0 → v1.12.1](https://github.com/woodruffw/zizmor-pre-commit/compare/v1.11.0...v1.12.1)
- [github.com/astral-sh/ruff-pre-commit: v0.12.7 → v0.12.11](https://github.com/astral-sh/ruff-pre-commit/compare/v0.12.7...v0.12.11)

neutrinoceros committed:
Merge pull request #18579 from astropy/pre-commit-ci-update-config

invalid-email-address committed:
Update minimum required version of astropy-iers-data

pllim committed:
Merge pull request #18581 from astropy/update-astropy-iers-data-pin-1756772043

Update minimum required version of astropy-iers-data

pllim committed:
DOC: Fix broken link by using HTTPS

ayshih committed:
Fix bug where autoscale_None() was unaware of an interval

eerovaher committed:
Remove usage of `lazyproperty` from `units`

In `units` the standard library `cached_property` is a drop-in
replacement for our own `lazyproperty`.

mhvk committed:
Remove usage of `lazyproperty` from `units` (#18591)

In `units` the standard library `cached_property` is a drop-in
replacement for our own `lazyproperty`.

larrybradley committed:
Merge pull request #18577 from Cadair/wcsaxes-fix_set_ticks_unit

Fix units error on `CoordinateHelper.set_ticks`

eerovaher committed:
Refactor `Generic._validate_unit()`

The updates reduce code duplication between `Generic._validate_unit()`
and `_ParsingFormatMixin._validate_unit()`.

pllim committed:
TST: Disable Gaussian1DKernel in pickle test

Co-authored-by: Eero Vaher <eero.vaher@fysik.lu.se>

eerovaher committed:
Simplify unit formatter `_validate_unit()` methods

The `detailed_exception` parameter, which controlled the verbosity of
error messages, can be removed. Dealing with the error messages outside
`_validate_unit()` is simple enough.

larrybradley committed:
Merge pull request #18596 from pllim/disable-gauss1d-kern-test

TST: Disable Gaussian1DKernel in pickle test due to flakiness

larrybradley committed:
Merge pull request #18590 from ayshih/norm_without_data

Fixed an image-normalization bug where the interval could be ignored when plotting

mhvk committed:
Merge pull request #18593 from eerovaher/generic_validate_units

Remove `detailed_exception` parameter from unit formatter `_validate_unit()` methods

eendebakpt committed:
Merge pull request #18599 from eendebakpt/numpy_dtype_deprecations_v0

Replace deprecated setting of dtype on numpy arrays in io.fits

bsipocz committed:
Merge pull request #18584 from pllim/fix-galex-url

DOC: Fix broken link by using HTTPS

eerovaher committed:
Refactor unit formatter `_validate_unit()` methods

The `"vounit"` and `"ogip"` unit formats have deprecated units. The code
for handling deprecated units in those two formatters is now replaced
with a common implementation in `_ParsingFormatMixin`.

The `"generic"` format does not have to worry about deprecated units, so
it does not benefit from the code that is being added to
`_ParsingFormatMixin`. In fact calling `super()._validate_unit(...)` in
`Generic._validate_unit()` would slow it down (slightly).

mhvk committed:
Ensure imperial.__all__ also contains 'enable'

mhvk committed:
Merge pull request #18604 from mhvk/units-imperial-enable-to-all

Ensure imperial.__all__ also contains 'enable'

nstarman committed:
Merge pull request #18603 from eerovaher/fewer_validate_unit-methods

Refactor unit formatter `_validate_unit()` methods

eerovaher committed:
Reveal invalid identifiers in unit module __all__

The new test reveals that `astropy.units.cds.__all__` contains `"°"`
even though that is not a valid identifier and therefore does not belong
in `__all__`.

eerovaher committed:
Improve validating unit module __all__ entries

The unit definition modules define so many objects that maintaining the
`__all__` for them manually is not feasible. Instead the vast majority
of entries are generated by `generate_dunder_all()`. The function now
checks that everything in its output is a valid identifier. At least in
`cds` this previously was not the case.

mhvk committed:
Merge pull request #18606 from eerovaher/unit-module-dunder-all-indentifiers

Only insert valid identifiers to unit definition module `__all__` lists

larrybradley committed:
Replace deprecated indent function with textwrap.indent

larrybradley committed:
Add FITS test for extra bytes after last HDU

pllim committed:
Merge pull request #18607 from larrybradley/remove-deprecated-indent

Replace deprecated indent() with textutils.indent() in io.fits warning

neutrinoceros committed:
RFC: fix the only `PTH211` violation in the entire codebase

mhvk committed:
Merge pull request #18573 from mhvk/units-auto-generate-unit-stubs

Auto-generate units stubs for wheels

pllim committed:
Merge pull request #18611 from neutrinoceros/config/rfc/pth211

RFC: fix the only `PTH211` violation in the entire codebase

nstarman committed:
cosmo: annotate types & fix docs (#18546)

nden committed:
Merge branch 'main' into modeling/tst/pt028

nden committed:
Merge pull request #18319 from neutrinoceros/modeling/tst/pt028

TST: avoid default values in a test function (`PT028`) in `modeling`

larrybradley committed:
Merge pull request #18317 from neutrinoceros/stats/tst/pt028

TST: avoid default values in test functions (`PT028`) in `stats`

eerovaher committed:
Reveal that unit module __all__ can include keywords

The new test reveals that `astropy.units.cds.__all__` contains `"as"`
even though that is a reserved keyword.

eerovaher committed:
Stop adding keywords to unit module __all__ lists

The output of `generate_dunder_all()` used for dynamically generating
unit definition module `__all__` lists no longer contains Python
keywords.

mhvk committed:
Merge pull request #18614 from eerovaher/unit-modules-dunder-all-keywords

Stop inserting keywords into unit definition module `__all__` lists

neutrinoceros committed:
MNT: upgrade ruff to 0.13.0

pllim committed:
Merge pull request #18296 from neutrinoceros/coordinates/tst/refine-too-broad-pytest.warns

TST: narrow a warning catch in `coordinates`' test suite (fix `PT030`)

pllim committed:
Merge pull request #18616 from neutrinoceros/mnt/upgrade-ruff-0.13

MNT: upgrade ruff to 0.13.0

eerovaher committed:
Allow handling warnings when writing deprecated units

The `"ogip"` and `"vounit"` formats have deprecated units. Converting
such units to strings has so far caused an unavoidable warning. A new
parameter of the `UnitBase.to_string()` and unit formatter `to_string()`
methods allows silencing that warning or raising an error instead. This
is analogous to the `parse_strict` parameter of unit parsing methods.

The `StrEnum` that controls the new behavior is defined in a new module
so that it could be imported without having to worry about import loops.

eerovaher committed:
Allow automatically converting deprecated units

The "vounit" and "ogip" formats have deprecated units. It is now
possible to instruct the unit formatters to try to replace deprecated
units when writing them to strings. If automatic replacement is not
possible the deprecated unit is used and a warning is emitted.

An alternative would have been to implement a new method in `UnitBase`
to replace deprecated units, which could be used like
`u.erg.convert_deprecated("vounit").to_string(format="vounit")`
but then the user would have to specify the unit format twice and such
repetition should be avoided.

mhvk committed:
Merge pull request #18586 from eerovaher/writing-deprecated-units

Allow controlling warnings when converting deprecated units to strings

neutrinoceros committed:
Merge pull request #18537 from neutrinoceros/doc/supported-python-versions-badge

taldcroft committed:
Merge pull request #18506 from hamogu/fix_6549

Find metadata in CDS readme even for gzipped files

eerovaher committed:
Remove unneeded code from `coordinates/sites.py`

`SiteRegistry.__contains__()` can be removed because `SiteRegistry`
can inherit its `__contains__()` implementation from `Mapping`.

The `SiteRegistry._loaded_jsondb` attribute created by
`SiteRegistry.from_json()` is not used anywhere.

mhvk committed:
Merge pull request #18626 from eerovaher/cleanup-sites

Remove unneeded code from `coordinates/sites.py`

cmarmo committed:
Make fitscheck verify all HDUs before listing errors. (#18574)

* Allow fitscheck to check the entire file before exiting.

* Add test

* Test all the lines, add changelog

* Add test with missing checksum

* restore previous messages, avoid catching warnings

* back to regex to fix test for windows

---------

Co-authored-by: Simon Conseil <s.conseil@ip2i.in2p3.fr>

HarshadaRaut234 committed:
BUG: Fix latex representation for DexUnit in astropy.units

mhvk committed:
BUG: fix double 2422 sign in latex string repr of logarithmic units

nstarman committed:
cosmo: consolidate private variables about neutrinos (#18629)

* refactor: consolidate neutrino information
* refactor: rename massivenu to has_massive_nu
* refactor: remove _nu_info.massivenu_mass
* refactor: rename nmass(ive/less)nu to n_mass(ive/less)_nu
* refactor: rename nneutrinos to n_nu

Signed-off-by: nstarman <nstarman@users.noreply.github.com>

mhvk committed:
Merge pull request #18627 from HarshadaRaut234/fix-dexunit-latex

Fix for latex representation for DexUnit in astropy.units

nstarman committed:
refactor: simplify utils

Signed-off-by: nstarman <nstarman@users.noreply.github.com>

nstarman committed:
refactor: group scipy compat

Signed-off-by: nstarman <nstarman@users.noreply.github.com>

neutrinoceros committed:
Merge pull request #18633 from nstarman/cosmo/scipy-compat

saimn committed:
Compute maximum absolute and relative differences reported by ImageDataDiff on the full arrays (#18451)

* compute max abs/rel differences on the full arrays

Fix #18413

* keep previous behavior

* address review comments

* handle edge cases with invalid values

* mask invalid before computing differences

pllim committed:
API: make delay_doc_updates context manager a noop under Python's optimized mode (#17572)

nstarman committed:
Restrict some CI to astropy repo (#18619)

Co-authored-by: nstarman <8949649+nstarman@users.noreply.github.com>

eerovaher committed:
Simplify regression tests for #6347

The regression tests added in #6347 ensured that the angular separations
returned by `search_around_3d()` and `search_around_sky()` are `Angle`
instances even if there are no matches, but learning that was only
possible by reading the issue on GitHub despite how verbose the tests
were.

mhvk committed:
Merge pull request #18637 from eerovaher/6347-regression-tests

Simplify regression tests for #6347

mhvk committed:
Merge pull request #18631 from nstarman/cosmo/utils/all_cls_vars

refactor: simplify cosmology utils

nstarman committed:
feat(cosmo): curvature component (#18232)

There is now a `cosmology` trait for attributes and methods related to the
global curvature.

jeffjennings committed:
Merge branch 'astropy:main' into coordinates_data
